mkosi/kmod.py

```diff
@@ -107,7 +109,7 @@ def filter_kernel_modules(
         globs = [normalize_module_glob(p) for p in include if not p.startswith("re:")]
 
         for m in modules:
-            rel = os.fspath(Path(*m.parts[5:]))
+            rel = os.fspath(m.relative_to(modulesd))
```

Comment:
You're still breaking the regex path by making this change without changing the regex logic to deal with /kernel.